### ISSUE TITLE
Why re-create most of next.js?

### ISSUE THREAD

**Issue by tashburn** [jaredpalmer/after.js#43]
Jared -- What are your thoughts on this?

https://github.com/zeit/next.js/issues/3582

**Comment by jaredpalmer** [comment]
why not?

**Comment by jaredpalmer** [comment]
I have been fetching data the way that After.js does for almost a year now with Razzle--so it felt like it was time abstract this technique into a 'framework.' 

After some early prototypes of After.js, I invited both Michael and Ryan to the repo to get some feedback. A few days later Ryan DM'd me that Next might be moving to React Router in the future. I reached out to Guillermo as well and he confirmed this was the case and that they planned to move Next.js to React Router in v5, but that it could be weeks away. So I continued tinkering with After.js.

After I got HMR and codesplitting working, I felt compelled to share it with everyone. To the extent that it "re-creates" Next.js, <insert-innovation-quote-here>.

As for why I do not contribute to Next.js.... I did in the early days. Then I moved to kyt, and then I built Razzle. Generally speaking, I don't spend my time fixing bugs or contributing to projects that I don't use day to day.
